### PR TITLE
Sync pod mod

### PR DESCRIPTION
**Pull request by michaelgugino**

### CODE REVIEW

File: roles/openshift_node_group/files/sync.yaml [hunk 1]
```diff
@@ -102,18 +102,30 @@ spec:
...
           while true; do
-            if ! oc extract "configmaps/${name}" -n openshift-node --to=/etc/origin/node --confirm --request-timeout=10s --config /etc/origin/node/node.kubeconfig "--token=$( cat /var/run/secrets/kubernetes.io/serviceaccount/token )"  > /dev/null; then
+            if ! oc extract "configmaps/${name}" -n openshift-node --to=/etc/origin/node/tmp --confirm --request-timeout=10s --config /etc/origin/node/node.kubeconfig "--token=$( cat /var/run/secrets/kubernetes.io/serviceaccount/token )"  > /dev/null; then
               echo "error: Unable to retrieve latest config for node" 2>&1
               sleep 15 &
               wait $!
               continue
             fi
+
+            KUBELET_HOSTNAME_OVERRIDE=$(cat /etc/sysconfig/KUBELET_HOSTNAME_OVERRIDE) || :
```
Comment (vrutkovs) [inline roles/openshift_node_group/files/sync.yaml]: An additional file is not required - there is `/etc/sysconfig/origin-node`

**Comment by vrutkovs** [comment]
I don't think modifying sync pod is a good idea - when cloudprovider is enabled changing `nodeName` might break things.

An alternative solution would be passing `--hostname-override` flag to the kubelet instead, so that node config map won't have to be modified. See https://github.com/vrutkovs/openshift-ansible/commit/6a61e4c00f56880b97edeb10d167f16fe2c2061a

**Comment by michaelgugino** [comment]
> I don't think modifying sync pod is a good idea - when cloudprovider is enabled changing nodeName might break things.

That's true.  Users shouldn't do this because it would probably break things.  This is opt-in behavior.

File: roles/openshift_node_group/files/sync.yaml [hunk 2]
```diff
@@ -102,18 +102,26 @@ spec:
             ) &
             break
           done
-
+          mkdir -p /etc/origin/node/tmp
```
Comment (michaelgugino) [inline roles/openshift_node_group/files/sync.yaml]: -p prevents error on existing.

File: roles/openshift_node_group/files/sync.yaml (hunk 1 above)
Comment (michaelgugino) [reply roles/openshift_node_group/files/sync.yaml]: Easier to not have to muck with the existing file which may or may not be under some user's automation already.

Comment (sdodson) [reply roles/openshift_node_group/files/sync.yaml]: @michaelgugino Was there ever something that actually created this file or was it expected to be user provided?

**Pull request merged**